src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,108 @@
  */
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] arrayList;
+    private int currentCapacity;
+    private int currentIndex;
+
+    public ArrayList() {
+        arrayList = (T[]) new Object[DEFAULT_CAPACITY];
+        currentCapacity = DEFAULT_CAPACITY;
+    }
 
     @Override
     public void add(T value) {
-
+        addElementToArray(value, currentIndex);
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > currentIndex) {
+            throw new ArrayIndexOutOfBoundsException(" Index above the size of Array");
+        }
+        sizeCheck(index);
+        if (index < currentIndex) {
+            System.arraycopy(arrayList, index, arrayList, index + 1, currentIndex + 1);
+        }
+        addElementToArray(value, index);
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            addElementToArray(list.get(i), currentIndex);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= currentIndex) {
+            throw new ArrayIndexOutOfBoundsException(("No element with index"));
+        }
+        return arrayList[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (currentIndex <= index) {
+            throw new ArrayIndexOutOfBoundsException("Index above Array size");
+        }
+        sizeCheck(index);
+        arrayList[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        T removedElement = get(index);
+        System.arraycopy(arrayList, index + 1, arrayList, index, currentIndex - 1);
+        currentIndex--;
+        return removedElement;
     }
 
     @Override
     public T remove(T t) {
-        return null;
+        for (int i = 0; i < currentIndex; i++) {
+            if (t == arrayList[i] || (t != null && t.equals(arrayList[i]))) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Such element does not exist.");
     }
 
     @Override
     public int size() {
-        return 0;
+        return currentIndex;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return currentIndex == 0;
+    }
+
+    public void addElementToArray(T value, int index) {
+        sizeCheck(index);
+        arrayList[index] = value;
+        currentIndex++;
+    }
+
+    public void increaseCapacity() {
+        currentCapacity = currentCapacity + (currentCapacity >> 1);
+        Object[] arrayListTemporary = (T[]) new Object[currentCapacity];
+        System.arraycopy(arrayList, 0, arrayListTemporary, 0, currentIndex);
+        arrayList = (T[]) arrayListTemporary;
+    }
+
+    public void sizeCheck(int sizeForTest) {
+        if (sizeForTest == currentCapacity) {
+            increaseCapacity();
+        }
```

Comment:
you can move this check to increaseCapacity() . And think if you really need to pass some parameters to this method to make this check